src/main/java/lol/fairplay/ghidraapple/analysis/objcclasses/OCClassFieldAnalyzer.kt

```diff
@@ -0,0 +1,167 @@
+package lol.fairplay.ghidraapple.analysis.objcclasses
+
+import ghidra.app.services.AbstractAnalyzer
+import ghidra.app.services.AnalysisPriority
+import ghidra.app.services.AnalyzerType
+import ghidra.app.util.importer.MessageLog
+import ghidra.program.model.address.AddressSetView
+import ghidra.program.model.address.GenericAddress
+import ghidra.program.model.data.CategoryPath
+import ghidra.program.model.data.*;
+import ghidra.program.model.data.Structure
+import ghidra.program.model.listing.Data
+import ghidra.program.model.listing.Program
+import ghidra.program.model.scalar.Scalar
+import ghidra.program.model.symbol.Namespace
+import ghidra.program.model.symbol.Symbol
+import ghidra.util.task.TaskMonitor
+
+private data class IVarField(val name: String, val type: String, val size: Int, val offset: Int)
+private data class IVarFieldList(val classSymbol: Symbol, val ivars: List<IVarField>)
+
+class OCClassFieldAnalyzer : AbstractAnalyzer(NAME, DESCRIPTION, AnalyzerType.DATA_ANALYZER) {
+
+    lateinit var program: Program
+    lateinit var log: MessageLog
+
+    companion object {
+        const val NAME = "Objective-C Class Field Analyzer"
+        const val DESCRIPTION = "write this later"
+        val PRIORITY = AnalysisPriority.DATA_ANALYSIS
+    }
+
+    init {
+        priority = PRIORITY
+        setPrototype()
+        setSupportsOneTimeAnalysis()
+    }
+
+    override fun canAnalyze(program: Program): Boolean {
+        this.program = program
+
+        val objc_const_sect = program.memory.getBlock("__objc_const")
+        return objc_const_sect != null
+    }
+
+    override fun added(program: Program, set: AddressSetView, monitor: TaskMonitor, log: MessageLog): Boolean {
+        this.log = log
+
+        val idDataType = program.dataTypeManager.getDataType("/_objc2_/ID")
+        val fieldLists = getIVarListsInAddressSet(set, monitor) ?: return false
+
+        monitor.message = "Applying class structure fields..."
+        monitor.progress = 0
+        monitor.maximum = fieldLists.size.toLong()
+
+        program.withTransaction<Exception>("Applying fields...") {
+            for (it in fieldLists) {
+                val definedClassStruct = tryResolveDefinedStruct(it.classSymbol.name) as Structure?
+                if (definedClassStruct == null) {
+                    log.appendMsg("Couldn't find defined structure for ${it.classSymbol.name} ivar list.")
+                    continue
+                }
+
+                it.ivars.forEach { field ->
+                    val fieldType = field.type.let {
+                        tryResolveDefinedStruct(it) ?: idDataType
+                    }
+
+                    // review: not quite sure if this sizing is adequate - adeluca
+                    val fieldSize = if (fieldType == idDataType) {
+                        8
+                    } else {
+                        field.size.toInt()
+                    }
```

Comment:
Ghidra kept giving me a warning related to the sizing of `ID`s that seemed to go away by making this change. I also remember that certain functions would not decompile because of an issue related to structure field offsets.